### PR TITLE
Wait for async saving and removing

### PR DESCRIPTION
**Pull request by vepasto**

### CODE REVIEW

**Comment by vepasto** [comment]
Now done is passed to exec as a parameter

**Comment by ilanbiala** [comment]
Can you fix the indentation and spacing to match the styles in the project? Spaces between `() {` is one thing, you should be able to see what changes to make based on JSHint.

**Comment by vepasto** [comment]
It seems to be already okay

```
Running "jshint:all" (jshint) task
>> 56 files lint free.
```

File: app/tests/article.server.model.test.js [hunk 1]
```diff
@@ -57,8 +57,8 @@ describe('Article Model Unit Tests:', function() {
 	});
 
 	afterEach(function(done) {
-		Article.remove().exec();
-		User.remove().exec();
-		done();
+		Article.remove().exec(function(){
```
Comment (ilanbiala) [inline app/tests/article.server.model.test.js]: space between `()` and `{`.

File: app/tests/user.server.model.test.js [hunk 2]
```diff
@@ -52,10 +52,11 @@ describe('User Model Unit Tests:', function() {
 		});
 
 		it('should fail to save an existing user again', function(done) {
-			user.save();
-			return user2.save(function(err) {
-				should.exist(err);
-				done();
+			user.save(function(){
```
Comment (ilanbiala) [inline app/tests/user.server.model.test.js]: space between `()` and `{`.

**Comment by vepasto** [comment]
Fixed formatting

**Comment by ilanbiala** [comment]
Same thing here, squash and then I'll merge it in.

**Comment by vepasto** [comment]
Squashed

**Pull request merged**